File: python/binary-search/binary_search.py
def find(search_list, value):
    if len(search_list) < 1 or \
        (len(search_list) == 1 and value != search_list[0]):
        raise ValueError('value not in array')

    mid_point = len(search_list) // 2

    if value == search_list[mid_point]:
        return mid_point

    if value < search_list[mid_point]:
        return find(search_list[:mid_point], value)

    if value > search_list[mid_point]:
        return mid_point + find(search_list[mid_point:], value)
    

File: javascript/binary-search/binary-search.js
export const find = (search_list, value) => {
    if (search_list.length < 1 ||
        (search_list.length == 1 && value != search_list[0])) {
      throw new Error('Value not in array')
  }

    const mid_point = Math.floor(search_list.length / 2)

    if (value == search_list[mid_point]) {
        return mid_point
    }

    if (value < search_list[mid_point]) {
        return find(search_list.slice(0, mid_point), value)
    }

    if (value > search_list[mid_point]) {
        return mid_point + find(search_list.slice(mid_point, search_list.length), value)
    }
}
